### PR TITLE
Add support for OS X and manylinux platform tags

### PR DESCRIPTION
**Pull request by njsmith**
Also removed "win-amd64" and "win-ia64" from the whitelist, since they
violate PEP 425.

Fixes: gh-1008
Fixes: gh-1009

### CODE REVIEW

**Comment by dstufft** [comment]
You have linting errors here.

**Comment by njsmith** [comment]
Fixed, tests are passing

File: warehouse/legacy/api/pypi.py [hunk 1]
```diff
@@ -786,11 +807,12 @@ def file_upload(request):
...
-            if set(plats) - ALLOWED_PLATFORMS:
-                raise _exc_with_message(
-                    HTTPBadRequest,
-                    "Binary wheel for an unsupported platform.",
-                )
+            for plat in plats:
+                if not _valid_platform_tag(plat):
+                    raise _exc_with_message(
+                        HTTPBadRequest,
+                        "Binary wheel for an unsupported platform.",
```
Comment (ogrisel) [inline warehouse/legacy/api/pypi.py]: I think it would be helpful for the users to report the offending paltform tag(s) in the exception message.

Comment (dstufft) [inline warehouse/legacy/api/pypi.py]: I agree, though that would probably require building a dictionary that maps platform tag to Valid/Invalid and then doing an all() on the values of that dictionary or something.

@njsmith If you don't mind making this change in this PR, I think that would be a great addition.

Comment (ogrisel) [inline warehouse/legacy/api/pypi.py]: My suggestion was just to do something like:

```python
raise _exc_with_message(
    HTTPBadRequest,
    "Binary wheel %s for an unsupported platform: %s" % (filename, plat),
)
```

**Comment by njsmith** [comment]
Improved error message and tests are passing -- ready to go I think.

**Pull request merged**